### DISCUSSION TITLE
The engine must run without installing Visual Studio!

### DISCUSSION THREAD

**Discussion by mahdisml** [o3de/o3de#15070] [Ideas]
The engine needs to install Visual Studio (C++) to run! 
Its Weird !
This is very annoying !
It must run without installing Visual Studio (C++).

**Comment by AMZN-alexpete** [comment]
Are you using the installer or did you download the source code?
Visual Studio (or some compiler) is needed to compile your project.  If you don't want to use Visual Studio, you should be able to use Clang or Ninja with Visual Studio build tools, and another possibility would be to use Linux.

Why can you not install Visual Studio?

**Reply by mahdisml** [reply]
I installed the editor with the installer.
My OS is Windows, and the editor forced me to install Visual Studio (and C++ build tools) with huge download sizes and storage space requirements for building my project!
I find it strange because other game engines use built-in tools to build projects.
This will bother the beginners.

**Comment by CodeLikeCXK** [comment]
You can still use it without VS.
You just need VS build tools.
I am using Rider for o3de

**Comment by razvanab** [comment]
I have just tried to build my project with only VS build tools installed and it doesn't work. 
It would be a good idea to explore the possibilities of having some sort of build environment packaged with the engine itself.

**Reply by CodeLikeCXK** [reply]
Basically if you don't have VS you may need to use command lines to create .sln files using Cmake

**Comment by wrstldev** [comment]
I found this thread via google search wondering about this exact issue.

> Why can you not install Visual Studio?

Because you have a two tiered system and it ceases to be "open" on Windows.

The Linux version requires free/open C++ tools and compilers to run.

But the Windows version requires expensive proprietary tools to be installed and used instead. "Visual Studio Personal Edition" has a restrictive license (which can change or cease to exist at any time) and is nothing more than an advertisement for the Professional and Enterprise editions. Linux teams develop for free, and Windows teams have to start paying expensive subscription fees as soon as they start working on a serious project.

Free / open tools should be standardized across platforms. No need to let Microsoft and Autodesk slip in and poison the well again.

**Reply by ShaunaGordon** [reply]
While I agree that standardizing on Clang and building that into the system more would be ideal -- in no small part because of the licensing issues of proprietary software -- your assessment of Visual Studio's licensing and the usage requirements isn't really accurate, which detracts from your argument.

Visual Studio is only required insofar as the build system is needed. This is actually almost identical to Mac users needing Xcode to build basically anything. And like Mac users and Xcode, you don't need to touch Visual Studio otherwise. This means that even if Community Edition (which is another thing - if you're looking at something that actually says "Personal Edition," you're either using a very old version or something that isn't actually VS) was "nothing more than an advertisement for the Professional and Enterprise editions" (which it's not and in fact has the same features as Professional Edition), you can still bypass it and use your tools of choice (and unless you're modifying the engine code, you're probably not going to need much outside of the O3DE editor anyway).

And regarding the expensive subscription fees, Microsoft doesn't consider a company an enterprise until they have more than 250 computers or are making over a million in annual revenue, both of which are a far cry from "as soon as they start working on a serious project."

Now, that could certainly change at some point without warning, I agree. That alone makes it worth standardizing on the open source options, since licensing changes by companies is a driving reason many of us use O3DE to begin with.

**Comment by loopaq** [comment]
> Are you using the installer or did you download the source code? Visual Studio (or some compiler) is needed to compile your project. If you don't want to use Visual Studio, you should be able to use Clang or Ninja with Visual Studio build tools, and another possibility would be to use Linux.
> 
> Why can you not install Visual Studio?

Hi am another o3de user who **do not**  like visual studio as a prerequisite for using o3de  without providing alternate workflow.

1. We can use linux (and we do use linux wherever we can use it ) but then we can't release our game to windows because o3de does not support cross compiling ,that would be  better solution but more work for o3de team to support cross compilation.
2. We cant use Visual studio because visual studio is proprietary and requires paid licence
3. it might be possible to build o3de from source(if through installer ,that would be best) using clang ninja and visual studio build tools but that workflow is not documentated .I have created an issue on github already for that here https://github.com/o3de/o3de.org/issues/2508
4. visual studio build tools also require a valid visual studio license unless you are building open source dependencies, as mentioned here https://devblogs.microsoft.com/cppblog/updates-to-visual-studio-build-tools-license-for-c-and-cpp-open-source-projects/

so how does it that license apply to game developers making games using o3de ,its confusing for newcomers, might scare them, will they end up paying for visual studio if they use visual studio build tools with o3de or they are safe?
That is a concern and must be addressed somewhere on the docs. o3de has visual scripting and lua for writing game scripts already then why we are forced to use visual studio , we would not even install visual studio if it was not mentioned as one of the requirements.  
An alternative workflow without visual studio is a must for o3de otherwise its a dealbreaker for many users.

**Comment by KennyAWT** [comment]
I hate visual studio , too heavy , also with paid license .... Why not with codeblocks and his compiler?

**Comment by childofdawn** [comment]
Any progress on this?

**Reply by Andre-LA** [reply]
It has been a while since you can use O3DE without Visual Studio, by using the script-only mode.

The advantage is that you can use O3DE without installing C++ toolchain, like VS.

The disadvantage is that you will be constrained to Lua and Script Canvas for game code, and the final build will not be as optimized as it could be by using C++ toolchain.

**Reply by childofdawn** [reply]
I see, thanks!
